engine/player/player.cpp

```diff
@@ -3029,6 +3029,123 @@ void player_t::init_finished()
       }
     } );
   }
+
+  if ( !precombat_state_map.empty() )
+  {
+    sim->error( "Warning: The 'override.precombat_state' option may not be fully supported for all buffs and cooldowns and and may produce incorrect or misleading results." );
+    std::unordered_map<buff_t*, std::tuple<int, double, timespan_t>> precombat_buff_state;
+
+    for ( const auto& v : precombat_state_map )
+    {
+      auto splits = util::string_split<std::string>( v.first, "." );
```

Comment:
I am fairly sure you could use the std::string_view version of util::string_split here.